### PR TITLE
used build for semver class

### PR DESCRIPTION
**Pull request by chillaq**
# Ruby SDK

### CODE REVIEW

File: lib/splitclient-rb/engine/matchers/semver.rb [hunk 1]
```diff
@@ -6,22 +6,33 @@ class Semver
...
-    # Class Initializer
-    #
-    # @param version [String] raw version as read from splitChanges response.
     def initialize(version)
       @major = 0
       @minor = 0
       @patch = 0
       @pre_release = []
       @is_stable = false
       @old_version = version
```
Comment (sanzmauro) [inline lib/splitclient-rb/engine/matchers/semver.rb:17]: remove `old_version`, it is not necessary now.

File: lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb [hunk 2]
```diff
@@ -10,20 +10,20 @@ def initialize(attribute, list_value, logger, validator)
...
 
-      value_to_match = args[:attributes][@attribute.to_sym]
-      unless value_to_match.is_a?(String)
+      value_to_match = SplitIoClient::Semver.build(args[:attributes][@attribute.to_sym], @logger)
+      unless !value_to_match.nil? && @semver_list.all? { |n| !n.nil? }
         @logger.log_if_debug('whitelistMatcherData is required for IN_LIST_SEMVER matcher type')
         return false
       end
-      matches = (@semver_list.map { |item| item.compare(SplitIoClient::Semver.new(value_to_match)) }).any?(&:zero?)
+      matches = (@semver_list.map { |item| item.compare(value_to_match) }).any?(&:zero?)
```
Comment (sanzmauro) [inline lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb:26]: don't use compare here. 
`item.version == value_to_match.version`

File: lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb [hunk 3]
```diff
@@ -10,20 +10,20 @@ def initialize(attribute, list_value, logger, validator)
       super(logger)
       @validator = validator
       @attribute = attribute
-      @semver_list = list_value.map { |item| SplitIoClient::Semver.new(item) }
+      @semver_list = list_value.map { |item| SplitIoClient::Semver.build(item, logger) }
```
Comment (sanzmauro) [inline lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb:13]: what happen if item is `null` or `empty` here ?

**Review by sanzmauro — CHANGES_REQUESTED** [review]

File: lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb (hunk 2 above)
Comment (sanzmauro) [reply lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb:26]: same in equal_to_semver_matcher

File: lib/splitclient-rb/engine/matchers/semver.rb (hunk 1 above)
Comment (chillaq) [reply lib/splitclient-rb/engine/matchers/semver.rb:17]: will do

File: lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb (hunk 2 above)
Comment (chillaq) [reply lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb:26]: will do

File: lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb (hunk 3 above)
Comment (chillaq) [reply lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb:13]: In match function it checks if any of the items is nil and return false:
      value_to_match = SplitIoClient::Semver.build(args[:attributes][@attribute.to_sym], @logger)
      unless !value_to_match.nil? && @semver_list.all? { |n| !n.nil? }
        @logger.log_if_debug('whitelistMatcherData is required for IN_LIST_SEMVER matcher type')
        return false
      end

**Review by sanzmauro — CHANGES_REQUESTED** [review]

File: lib/splitclient-rb/engine/matchers/equal_to_semver_matcher.rb [hunk 4]
```diff
@@ -23,7 +23,7 @@ def match?(args)
         @logger.log_if_debug('stringMatcherData is required for EQUAL_TO_SEMVER matcher type')
         return false
       end
-      matches = @semver.compare(SplitIoClient::Semver.new(value_to_match)).zero?
+      matches = (@semver.version == SplitIoClient::Semver.new(value_to_match).version)
```
[the remark this answers is not in the document]
Comment (chillaq) [reply lib/splitclient-rb/engine/matchers/equal_to_semver_matcher.rb]: I will update all matchers with validation for nil.

**Review by sanzmauro — APPROVED** [review]

**Pull request merged**